### PR TITLE
Assignment 1: completed answers to all questions

### PR DESCRIPTION
**Pull request by HimeshiS**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Answered all questions in assignment-1 on KNN classification

## What did you learn from the changes you have made?
I learned how to apply KNN classification on datasets. I learned how to split datasets into training and test sets. I also learned how to implement a GridSearch to find the best value for k neighbors when you are unsure of what k should be.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
When choosing the best value for k from the grid search, I also could have plotted the results to visually inspect where the best k value would be. Often, with increasing k, the accuracy of predictions could go up but only slightly. There will be a trade-off between slight gains in accuracy at the expense of overfitting. 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
The challenge was to fully grasp the concept of KNN classification as it was a new concept to me. I went through the notebooks we covered in class and also did literature searches on my own to get a deeper understanding and complete the assignment.

## How were these changes tested?
N/A

## A reference to a related issue in your repository (if applicable)
N/A

## Checklist
- [X] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by amanda-ng518 — APPROVED** [review]
A1 PASS
